TRACING-3513 | Document that OTEL operator can create service monitors (#10)
max-cx/openshift-docs: modules/distr-tracing-otel-config-send-metrics-monitoring-stack.adoc

[id="distr-tracing-send-metrics-monitoring-stack_{context}"]
= Sending metrics to the monitoring stack

> The OpenTelemetry Collector custom resource can be configured to create Prometheus `ServiceMonitor` to scrape collector's pipeline metrics and enabled Prometheus exporters.

Reviewer: Question 1: The `ServiceMonitor` is what, a CR or a what exactly? Let's define it here (to avoid getting asked to do this by the merge reviewer).

Question 2: Is it `the enabled Prometheus exporters` or `enable the Prometheus exporters`?

```suggestion
You can configure the OpenTelemetry Collector custom resource to create a Prometheus `ServiceMonitor` to scrape the collector's pipeline metrics and the enabled Prometheus exporters.
```

Author: > the enabled Prometheus exporters

this one

>  The ServiceMonitor is what, a CR or a what exactly?

it is CR, how should I define it? It is universally known CR that configures prometheus to scrape metrics from pods or services

Reviewer: Thank you for your reply. This is sufficient. Let me have a look.

> You can configure the OpenTelemetry Collector custom resource to create a Prometheus `ServiceMonitor` to scrape the collector's pipeline metrics and the enabled Prometheus exporters.

Reviewer: ```suggestion
You can configure the OpenTelemetry Collector custom resource (CR) to create a Prometheus `ServiceMonitor` CR to scrape the collector's pipeline metrics and the enabled Prometheus exporters.
```

> .Example of the OpenTelemetry Collector custom resource with Prometheus exporter

Reviewer: ```suggestion
.Example of the OpenTelemetry Collector custom resource with the Prometheus exporter
```

[source,yaml]
----
spec:
  mode: deployment
  observability:
    metrics:
      enableMetrics: true <1>
  config: |
    exporters:
      prometheus:
        endpoint: 0.0.0.0:8889
        resource_to_telemetry_conversion:
          enabled: true # by default resource attributes are dropped
    service:
      telemetry:
        metrics:
          address: ":8888"
      pipelines:
        metrics:
          receivers: [otlp]
          exporters: [prometheus]
----

> <1> Configures the operator to create Prometheus `ServiceMonitor` to scrape collector's internal and Prometheus exporter metric endpoints. The metrics will be stored in the OpenShift monitoring stack.

Reviewer: ```suggestion
<1> Configures the operator to create the Prometheus `ServiceMonitor` CR to scrape the collector's internal and Prometheus exporter metric endpoints. The metrics will be stored in the OpenShift monitoring stack.
```

Reviewer: Also, could you please clarify: `to scrape the collector's internal [what exactly? metrics or metric endpoints?]`.

Author: added some words to text

spec:
  selector:
    matchLabels:

>       app.kubernetes.io/name: otel-collector <1>

Reviewer: In connection to my suggestion in the next thread below:
```suggestion
      app.kubernetes.io/name:  `<cr-name>-collector` <1>
```

    - action: labeldrop
      regex: job
----

> <1> The name of the OpenTelemetry customer resource e.g. `<cr-name>-collector`.

Reviewer: I suggest moving the `<cr-name>-collector` example into the code block. See my previous suggestion.

```suggestion
<1> The name of the OpenTelemetry custom resource.
```
